javatests/arcs/core/analysis/ClaimDeductionTest.kt

```diff
@@ -0,0 +1,211 @@
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.Claim
+import arcs.core.data.EntityType
+import arcs.core.data.FieldType
+import arcs.core.data.HandleConnectionSpec
+import arcs.core.data.HandleMode
+import arcs.core.data.ParticleSpec
+import arcs.core.data.Recipe
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.data.SingletonType
+import arcs.core.data.expression.PaxelParser
+import arcs.core.testutil.assertThrows
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+private fun accessPathOf(
+  particle: Recipe.Particle,
+  connection: HandleConnectionSpec,
+  vararg selectors: String
+) = AccessPath(particle, connection, selectors.map { AccessPath.Selector.Field(it) })
+
+@RunWith(JUnit4::class)
+class ClaimDeductionTest {
+  private val inputPetConnection = HandleConnectionSpec(
+    name = "input",
+    direction = HandleMode.Read,
+    type = SingletonType(EntityType(Schema(
+      names = setOf(SchemaName("PetCount")),
+      fields = SchemaFields(
+        singletons = mapOf(
+          "cat" to FieldType.Number,
+          "dog" to FieldType.Number,
+          "foo" to FieldType.Text
+        ),
+        collections = emptyMap()
+      ),
+      hash = "FooHousePetHash"
+    )))
+  )
+
+  private val outputPetConnection = HandleConnectionSpec(
+    name = "output",
+    direction = HandleMode.Write,
+    type = SingletonType(EntityType(Schema(
+      names = setOf(SchemaName("Foo")),
+      fields = SchemaFields(
+        singletons = mapOf(
+          "a" to FieldType.InlineEntity("BarHash"),
+          "b" to FieldType.Text,
+          "c" to FieldType.Number
+        ),
+        collections = emptyMap()
+      ),
+      hash = "FooHousePetHash"
+    ))),
+    expression = PaxelParser.parse(
```

Comment:
Set this to null here and define the expression explicitly in every test unless you are using this expression in multiple tests.